zuul-io-uring-test/src/test/java/com/netflix/zuul/netty/server/IoUringTest.java

```diff
@@ -0,0 +1,141 @@
+        logger.info("IOUring.isAvailable: {}", IOUring.isAvailable());
+        logger.info("IS_OS_LINUX: {}", IS_OS_LINUX);
+
+        if (IS_OS_LINUX) {
+            if (IOUring.unavailabilityCause() != null) {
+                logger.info(ExceptionUtils.getStackTrace(IOUring.unavailabilityCause()));
+            }
+            exerciseIoUringServer();
+        }
+    }
+
+    private void exerciseIoUringServer() throws Exception {
+        IOUring.ensureAvailability();
+
+        ServerStatusManager ssm = mock(ServerStatusManager.class);
+
+        Map<NamedSocketAddress, ChannelInitializer<?>> initializers = new HashMap<>();
+        final AtomicInteger ioUringChannelCount = new AtomicInteger(0);
+        ChannelInitializer<Channel> init = new ChannelInitializer<Channel>() {
+            @Override
+            protected void initChannel(Channel ch) {
+                logger.info("Channel: " + ch.getClass().getName());
+                logger.info("Channel isActive: " + ch.isActive());
+                logger.info("Channel isOpen: " + ch.isOpen());
+                assertTrue("isActive", ch.isActive());
+                assertTrue("isOpen", ch.isOpen());
+                if (ch instanceof IOUringSocketChannel) {
+                    ioUringChannelCount.incrementAndGet();
+                }
+            }
+        };
+        initializers.put(new NamedSocketAddress("test", new InetSocketAddress(0)), init);
+        // The port to channel map keys on the port, post bind. This should be unique even if InetAddress is same
+        initializers.put(new NamedSocketAddress("test2", new InetSocketAddress( 0)), init);
+
+        ClientConnectionsShutdown ccs =
+                new ClientConnectionsShutdown(
+                        new DefaultChannelGroup(GlobalEventExecutor.INSTANCE),
+                        GlobalEventExecutor.INSTANCE,
+                        /* discoveryClient= */ null);
+        EventLoopGroupMetrics elgm = new EventLoopGroupMetrics(Spectator.globalRegistry());
+        EventLoopConfig elc = new EventLoopConfig() {
+            @Override
+            public int eventLoopCount() {
+                return 1;
+            }
+
+            @Override
+            public int acceptorCount() {
+                return 1;
+            }
+        };
+        Server s = new Server(new NoopRegistry(), ssm, initializers, ccs, elgm, elc);
+        s.start();
+
+        List<NamedSocketAddress> addresses = s.getListeningAddresses();
+        assertEquals(2, addresses.size());
+
+        addresses.forEach(address -> {
+            assertTrue(address.unwrap() instanceof InetSocketAddress);
+            InetSocketAddress inetAddress = ((InetSocketAddress) address.unwrap());
+            assertNotEquals(inetAddress.getPort(), 0);
+            openHttpConnection(inetAddress.getPort());
+        });
+
+        s.stop();
+
+        assertEquals(2, ioUringChannelCount.get());
+    }
+
+    private static void openHttpConnection(final int port) {
```

Comment:
This doesn't actually validate a proper request-response?

Reply:
I replaced `openHttpConnection` with `checkConnection`